### PR TITLE
Add Gyre Archive Flag

### PR DESCRIPTION
**Pull request by TriumphantBass**
### Gyre Archives
- Kobo and Merchant Crow added as gating items in logic
- Bringing Kobo to the portrait room in the library spawns a portal to the Ifrit boss, with three checks
- Bringing Merchant Crow to the Historical Documents room in the lab spawns a portal to the Ravenlord boss, with three checks
- Interactions added in marked locations when Kobo/Merchant Crow not required to show requirements in game

## Other Changes
- Gyre locations are visible on map. Checks are visible when applicable.
- Gyre entrance is open, and exit redirects to entrance instead of a Gyre boss
- 3 item checks along the Gyre were added (Gyre has same entrance requirements as Nightmare itself, and thus cannot contain progression items unless in Multiworld
- Kobo and Merchant Crow added to the item tracker
- Readme updates

### CODE REVIEW

File: TsRandomizer/LevelObjects/Other/GyrePortalEvent.cs [hunk 1]
```diff
@@ -9,7 +9,36 @@ class GyrePortalEvent : LevelObject
...
+				LevelReflected.SetLevelSaveInt("GyreDungeonSeed", 0); // Warp to Ravenlord
+			}
+			// Backer room
+			else if (typedObject.Level.ID == 2 && typedObject.Level.RoomID == 51)
+            {
+				Dynamic._portalType = 2; // post-dungeon
+				LevelReflected.SetLevelSaveInt("GyreDungeonSeed", 1); // Warp to Ifrit
+			}
+
+			// TODO (currently have auto-warps on room entry)
```
Comment (JarnoWesthof) [inline TsRandomizer/LevelObjects/Other/GyrePortalEvent.cs:29]: These dead lines of code should be removed

File: ReadMe.md [hunk 2]
```diff
@@ -50,11 +51,11 @@ Extract the contents of the corresponding zip file to the same folder as Timespi
 * In Seed selection menu, Ctr+C can now be used to copy seed id to the clipboard
 
 # Routing changes
-* The drawbridge is now open by default, and doesn't require you to visit the twins demons
+* The drawbridge is now open by default, and doesn't require you to visit the demon twins
 * Amadeus Laboratory now requires keycard B to open
 * To prevent players from getting stuck in the past when items are still available in the present, the transition room next to the refugee camp can now be used to freely travel back and forth between past and present
 * When you obtain the pyramid keys, a random transition room is unlocked that could open new routing opportunities
-* The temporal gyre is closed, due to its high randomness inside an already randomized timezone, the timezone would be highly unstable
+* The Temporal Gyre's boss rooms are disconnected from its main loop, which instead exits to its entrance. When using the "Gyre Archives" flag, Ifrit can be reached by aquiring Kobo and proceeding to the portrait room in the Library, while Ravenlord can be reached by aquiring Merchant Crow and proceeding to the room in Amadeus' Library containing the Historical Archives
```
Comment (JarnoWesthof) [inline ReadMe.md]: Are you sure its called Amadeus' Library? i though it was someting like labetory?

File: TsRandomizer/LevelObjects/Other/HistoricalDocuments.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace TsRandomizer.LevelObjects.Other
+{
+	[TimeSpinnerType("Timespinner.GameObjects.Events.EnvironmentPrefabs.L11_Lab.EnvPrefabLabHistoricalDocuments")]
+	// ReSharper disable once UnusedMember.Global
+	class HistoricalDocuments: LevelObject
+	{
+		public HistoricalDocuments(Mobile typedObject) : base(typedObject)
+		{
+			TimeSpinnerGame.Localizer.OverrideKey("q_ram_4_lun_29alt",
```
Comment (JarnoWesthof) [inline TsRandomizer/LevelObjects/Other/HistoricalDocuments.cs:12]: Should probably check seed options here to see if the string needs to be overridden

**Review by JarnoWesthof — CHANGES_REQUESTED** [review]

**Comment by TriumphantBass** [comment]
Feedback addressed

File: TsRandomizer/LevelObjects/Other/HistoricalDocuments.cs (hunk 3 above)
Comment (TriumphantBass) [reply TsRandomizer/LevelObjects/Other/HistoricalDocuments.cs]: Added

File: TsRandomizer/LevelObjects/Other/GyrePortalEvent.cs (hunk 1 above)
Comment (TriumphantBass) [reply TsRandomizer/LevelObjects/Other/GyrePortalEvent.cs]: Removed

**Review by JarnoWesthof — APPROVED** [review]

**Pull request merged**